$\cos$

In [2]:
import numpy as np
import math

In [ ]:
def error_rule_1(constant, error):
    return constant*error
    

In [4]:
# constants is a dict where values have key relating to location Q = cA^m and error is uncertainy A
def error_rule_2(constants, error):
    c = constants
    return abs(c['c']*c['m']*c['A']**(c['m']-1))*error

In [3]:
# constants dict wehre key is name using format Q = A + B .. + Z or Q = A - B
# errors is dict where key is name uses same format above and value is uncertainy 
def error_rule_3(constants, error):
    if len(constants) != len(error):
        print('error dicts are required to be same length')
        return
    
    total_sqrt = 0
    for key, value in error.items():
        total_sqrt += value**2
    
    return math.sqrt(total_sqrt)

        
    

In [49]:
# Q = cA^m * B^n
# variables is a dict key = variable in above equation, with values [value, power_value, error]
def error_rule_4(const, variables):
    total = const
    error = 0
    for key, value in variables.items():
        # solving equation
        total *= (value[0]**value[1])
        # solving for part of uncertainy within sqrt
        error += ((value[1] * value[2] / value[0]) ** 2)
    error = math.sqrt(error)
    return total, total*error

In [50]:
# method one error calculation
const = math.sqrt(2*9.8)
variables_dict_1 = {}
variables_dict_1['m+M'] = [(257.4/1000), 1, (.25/1000)]
variables_dict_1['m'] = [(66/1000), -1, (.1/1000)]
variables_dict_1['R'] = [(27.5/100), .5, (.25/100)]
variables_dict_1['1-cos(theta)'] = [0.16371, .5,.0007]
print(error_rule_4(const, variables_dict_1))

(3.6635033573070452, 0.019547768946673125)


Calculated values in *method 1* match calculated values in function

In [51]:
const = math.sqrt(9.8/2)
variables_dict_2 = {}
variables_dict_2['x'] = [(160.26/100), 1, (1.3/100)]
variables_dict_2['y'] = [(98.9/100), -.5, (.5/100)]
print(error_rule_4(const, variables_dict_2))

(3.5671800675006176, 0.030308731494307455)


Calculated values in *method 2* match calculated values in function

In [48]:
# USING PYTHON FOR CALCULATIONS
x = [1.1,1.3,1.4,0.9,0.95, 1.05]
print('uncertainy', np.std(x)/(math.sqrt(len(x))))
print('average', np.average(x))

uncertainy 0.073282810879294
average 1.1166666666666667


We found the average of x to be $1.12$ and we calculated the uncertainy using $$\frac{\sigma}{\sqrt{6}} $$ which we found to be $0.07$

Rule 3:
if $ Q = cA^m $
then
$$ \delta Q = cmA^{m-1}\delta A $$

We calculate $v_i$ in method 2 of the previous lab using the equation below
$$ v_i = \Delta x \sqrt{\frac{g}{2\Delta y}} $$

to calculate error in $v_i$
$$ \delta v_i = v_i\sqrt{\left(\frac{1*\delta (m+M)}{m+M}\right)^2 +\left(\frac{-1*\delta (m)}{m}\right)^2 +\left(\frac{\frac{1}{2}*\delta (R)}{R}\right)^2 +\left(\frac{\frac{1}{2}*\delta (1 - \cos \theta)}{(1 - \cos \theta)}\right)^2}$$
